tests/test_qnode.py

```diff
@@ -923,6 +923,176 @@ def circuit(a, b):
+            np.sin(a)**2,
+            np.cos(a)*np.cos(b),
+            0.25*(3-2*np.cos(b)**2*np.cos(2*c)-np.cos(2*b))
+        ])
+        assert np.allclose(var, expected, atol=tol, rtol=0)
+
+        # # circuit jacobians
+        gradA = circuit.jacobian([a, b, c], method='A')
+        gradF = circuit.jacobian([a, b, c], method='F')
+        expected = np.array([
+            [2*np.cos(a)*np.sin(a), -np.cos(b)*np.sin(a), 0],
+            [0, -np.cos(a)*np.sin(b), 0.5*(2*np.cos(b)*np.cos(2*c)*np.sin(b)+np.sin(2*b))],
+            [0, 0, np.cos(b)**2*np.sin(2*c)]
+        ]).T
+        assert np.allclose(gradF, expected, atol=tol, rtol=0)
+        assert np.allclose(gradA, expected, atol=tol, rtol=0)
+
+    def test_first_order_cv(self, tol):
+        """Test variance of a first order CV expectation value"""
+        dev = qml.device('default.gaussian', wires=1)
+
+        @qml.qnode(dev)
+        def circuit(r, phi):
+            qml.Squeezing(r, 0, wires=0)
+            qml.Rotation(phi, wires=0)
+            return qml.var(qml.X(0))
+
+        r = 0.543
+        phi = -0.654
+
+        var = circuit(r, phi)
+        expected = np.exp(2*r)*np.sin(phi)**2 + np.exp(-2*r)*np.cos(phi)**2
+        assert np.allclose(var, expected, atol=tol, rtol=0)
+
+        # circuit jacobians
+        gradA = circuit.jacobian([r, phi], method='A')
+        gradF = circuit.jacobian([r, phi], method='F')
+        expected = np.array([
+            2*np.exp(2*r)*np.sin(phi)**2 - 2*np.exp(-2*r)*np.cos(phi)**2,
+            2*np.sinh(2*r)*np.sin(2*phi)
+        ])
+        assert np.allclose(gradA, expected, atol=tol, rtol=0)
+        assert np.allclose(gradF, expected, atol=tol, rtol=0)
+
+    def test_second_order_cv(self, tol):
+        """Test variance of a second order CV expectation value"""
+        dev = qml.device('default.gaussian', wires=1)
+
+        @qml.qnode(dev)
+        def circuit(n, a):
+            qml.ThermalState(n, wires=0)
+            qml.Displacement(a, 0, wires=0)
+            return qml.var(qml.MeanPhoton(0))
+
+        n = 0.12
+        a = 0.765
+
+        var = circuit(n, a)
+        expected = n ** 2 + n + np.abs(a) ** 2 * (1 + 2 * n)
+        assert np.allclose(var, expected, atol=tol, rtol=0)
+
+        # circuit jacobians
+        gradF = circuit.jacobian([n, a], method='F')
+        expected = np.array([2*a**2+2*n+1, 2*a*(2*n+1)])
+        assert np.allclose(gradF, expected, atol=tol, rtol=0)
+
+    def test_error_analytic_second_order_cv(self):
+        """Test exception raised if attempting to use a second
+        order observable to compute the variance derivative analytically"""
+        dev = qml.device('default.gaussian', wires=1)
+
+        @qml.qnode(dev)
+        def circuit(a):
+            qml.Displacement(a, 0, wires=0)
+            return qml.var(qml.MeanPhoton(0))
+
+        with pytest.raises(ValueError, match=r"cannot be used with the parameter\(s\) \{0\}"):
+            circuit.jacobian([1.], method='A')
+
+
```

Comment:
Should still work! We have the analytic formula for variance of 2nd order observables. Even the gradient will still compute, it will just fallback to finite differences

Reply:
See `test_second_order_cv` above.